docs/schema-conversion-logic.md

```diff
@@ -0,0 +1,98 @@
+# Schema conversion logic
+
+Metrictank comes with a schema conversion logic (`mdata/importer/converter.go`) which can be used to convert data from one storage schema into another one. This is currently only used by the whisper importer, but may in the future also be used to import from other types of data sources.
+
+The conversion happens on a destination archive by destination archive basis. In this context the term "archive" refers to one section between the `,` of a storage schema, f.e. the schema `1s:1d,10min:1y` has two archives. Each destination archive gets generated separately, in some cases the same input archives may be used to generate multiple destination archives and in others different input archives may be used. Multiple input archives may also be combined into a single destination archive, if this makes sense given the input schema and the destination schema.
+
+This is how the schema conversion logic generates a destination archive:
+
+  * In the set of source archives which it has received as input, it finds the archive with the shortest retention which still contains enough data to satisfy the retention of the destination archive
+  * In the same set of source archives, it finds the archive with the longest retention that still has a higher or equal resolution as the destination archive
```

Comment:
similarly here, will it pick the 2nd one? the first one seems better suited
```
dest: 4h:1y
src: 2h:2y,3h:3y
```

Reply:
I agree that if the high res one has enough data, then the result is going to be better if we use it to re-generate the rollup with the adjusted aggregation span. but at the moment it would use the 2nd one and generate an approximation based on it. so this documents the correct behavior, which we can change in the future with another PR

Reply:
ok, does your existing PR fix it? link to it here. or does it not? -> can you create a new ticket?

Reply:
https://github.com/grafana/metrictank/issues/1609